### DISCUSSION TITLE
Loading esp-01's in bulk

### DISCUSSION THREAD

**Discussion by V1EngineeringInc** [luc-github/ESP3D#822] [Questions]
Hi,

Love the project. I am looking to speed up loading 100-200 ESP-01's so some of my users can give ESP3D/ESPUi a try.

I loaded up V3, did some config changes, macro's, and ran some tests. So now I tried to use platformio to build and load the filesystem that I added to esp3d/data. It shows a success but no files get loaded.

`|-- ESP8266-Arduino-Lua @ 0.0.30
|-- ESP8266mDNS @ 1.2
|   |-- ESP8266WiFi @ 1.0
|-- ArduinoOTA @ 1.0
|   |-- ESP8266WiFi @ 1.0
|   |-- ESP8266mDNS @ 1.2
|   |   |-- ESP8266WiFi @ 1.0
|-- DNSServer @ 1.1.1
|   |-- ESP8266WiFi @ 1.0
|-- ESP8266HTTPClient @ 1.2
|   |-- ESP8266WiFi @ 1.0
|-- BMx280MI @ 1.2.0
|   |-- Wire @ 1.0
|   |-- SPI @ 1.0
|-- DHT sensor library for ESPx @ 1.0.6
|-- WebSockets @ 2.3.6
|   |-- Hash @ 1.0
|   |-- ESP8266WiFi @ 1.0
|   |-- Ethernet @ 2.0.0
|   |   |-- SPI @ 1.0
|   |-- SPI @ 1.0
Building in release mode
Building file system image from 'esp3d\data' directory to .pioenvs\esp8266_1MB_wifi\spiffs.bin
/esp32cam.html
/export.json
/favicon.ico
/index.html.gz
/preferences.json
/tool.html
Looking for upload port...
Auto-detected: COM4
Uploading .pioenvs\esp8266_1MB_wifi\spiffs.bin
esptool.py v3.0
Serial port COM4
Connecting....
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: e8:68:e7:cb:d7:b0
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Compressed 262144 bytes to 114016...
Writing at 0x000bb000... (14 %)
Writing at 0x000bf000... (28 %)
Writing at 0x000c3000... (42 %)
Writing at 0x000c7000... (57 %)
Writing at 0x000cb000... (71 %)
Writing at 0x000cf000... (85 %)
Writing at 0x000d3000... (100 %)
Wrote 262144 bytes (114016 compressed) at 0x000bb000 in 10.0 seconds (effective 208.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...`

Is this a bug or am I doing it wrong? Is there any other way to flash and load the files in one shot? I would really like to speed up the loading process.

Thanks a million, 
Ryan

**Comment by luc-github** [comment]
v3 does not use spiffs but litllefs it seems you use the wrong tool.in platformio

**Reply by V1EngineeringInc** [reply]
Could you give me a little nudge in the right direction? 

I used the build file system button then the upload file system button in the Platform.io tab. What is the correct way?

**Reply by luc-github** [reply]
I remember need to add entry in platformio to define the filesystem
https://docs.platformio.org/en/stable/platforms/espressif8266.html#using-filesystem

**Reply by luc-github** [reply]
`board_build.filesystem = littlefs` it help to select right tool I think
Be sure the data directory use latest webUI and content fit the FS partition on ESP01 you have limited space if you use 1M ESP01 board

**Reply by V1EngineeringInc** [reply]
Thank you! I will try it as soon as possible.

**Reply by luc-github** [reply]
Let me know - if ok I can add it to configurator also https://luc-github.github.io

**Reply by V1EngineeringInc** [reply]
Thanks Luc that does work. The files now load.

The problem is they are not used. The preferences, favicon, export.json, are now on the esp-01 but I still need to log in and load the Export.json to get the those settings loaded the rest is good. Is there a way to build it in somehow?

**Reply by V1EngineeringInc** [reply]
Sorry I edited the post.

**Reply by luc-github** [reply]
export.json is only to transfer data  but passwords
what do you put in export.json that is not in preferences.json ?

**Reply by V1EngineeringInc** [reply]
Different SSID, host name, Baud rate, and target firmware.

**Reply by V1EngineeringInc** [reply]
Isn't that all the settings on the features tab?

**Comment by V1EngineeringInc** [comment]
I know this seems like kind of a small thing but the new boards I have coming use a ESP32 and I flash a handful of boards everyday. It sure makes things nice when I can set them up once, and just flash each new order one time and test. It saves me a ton of time to have all the settings available. Any changes I make can easily be shared or updated with all the user now and in the future as well. A lot of my users are not comfortable with the computer side of things so I go out of my way to make it as easy as possible for everyone.

**Reply by luc-github** [reply]
if it is done on your side there is another solution : you do your master ESP32 or ESP8266 with all changes you need including password,  and dump the full image using esptool (4MB), it will give you a 4MB binary file image, then you can flash the full image using esptool or any other tool to your new esp

**Reply by luc-github** [reply]
Just be sure you flash ESP32 image on ESP32 board and ESP8266 image on ESP8266 board

**Reply by V1EngineeringInc** [reply]
Perfect!!!! Thank you, that is what I needed.
